samples/cpp/classification_sample_async/main.cpp

```diff
@@ -170,7 +170,7 @@ int main(int argc, char* argv[]) {
         // -------- Step 10. Do asynchronous inference --------
         infer_request.set_callback([&](std::exception_ptr ex) {
             if (ex)
-                throw ex;
+                std::rethrow_exception(ex);
```

Comment:
You need to propagate the exception to the main thread

Reply:
@Wovchena could you suggest what exactly should be done?

Reply:
Same try catch as in https://github.com/openvinotoolkit/open_model_zoo/pull/3376

Reply:
Please use `std::promise<void>/std::future<void>` to wait for execution or propagate exception to main thread. 
Please use `OVInferRequestCallbackTests.canStartSeveralAsyncInsideCompletionCallbackWithSafeDtor` as example

Reply:
@apankratovantonp , please see current variant using variable for checking status of exception. Sample gets error and not hanging without std::future

Reply:
@apankratovantonp or do you want to realize it using std::future instead of current solution?

Reply:
The last solution is enough to propagate exception to main thread.